pageserver/src/tenant.rs

```diff
@@ -4258,6 +4261,53 @@ mod tests {
         tline.freeze_and_flush().await.map_err(|e| e.into())
     }
 
+    #[tokio::test]
+    async fn test_lsn_leases() -> anyhow::Result<()> {
+        let (tenant, ctx) = TenantHarness::create("test_lsn_leases")?.load().await;
+        let key = Key::from_hex("010000000033333333444444445500000000").unwrap();
+        let end_lsn = Lsn(0x50);
+        let timeline = tenant
+            .create_test_timeline_with_layers(
+                TIMELINE_ID,
+                Lsn(0x10),
+                DEFAULT_PG_VERSION,
+                &ctx,
+                Vec::new(),
+                vec![
+                    (Lsn(0x20), vec![(key, test_img("data key 0"))]),
+                    (Lsn(0x30), vec![(key, test_img("data key 1"))]),
+                    (Lsn(0x40), vec![(key, test_img("data key 2"))]),
+                ],
+                end_lsn,
+            )
+            .await?;
+
+        let _ = timeline.make_lsn_lease(Lsn(0x20), &ctx)?;
+
+        // Force set disk consistent lsn so we can get the cutoff at `end_lsn`.
+
+        timeline.force_set_disk_consistent_lsn(end_lsn);
+
+        let res = tenant
+            .gc_iteration(
+                Some(TIMELINE_ID),
+                0,
+                Duration::ZERO,
+                &CancellationToken::new(),
+                &ctx,
+            )
+            .await?;
+
+        // Keeping everything <= Lsn(0x20) b/c leases: {0/10, 0/20};
+        assert_eq!(res.layers_needed_by_leases, 2);
```

Comment:
could you please add a comment on why there are two layers? (i.e., one is the image layer we added when init the timeline and one is the initdb layer)

Reply:
addressed